Dockerfile

```diff
@@ -17,7 +17,26 @@ WORKDIR /todo-app/todo_app
 
 ENTRYPOINT poetry run gunicorn wsgi:start -b 0.0.0.0:80
 
-FROM base as development
+FROM base as dev_base
 RUN poetry install
 
-ENTRYPOINT poetry run flask run --host=0.0.0.0
\ No newline at end of file
+FROM dev_base as development
+ENTRYPOINT poetry run flask run --host=0.0.0.0
+
+FROM dev_base as test
+ 
+# Install the long-term support version of Firefox (and curl if you don't have it already)
+RUN apt-get install -y firefox-esr curl
```

Comment:
I'd run apt-get update before this :) To make sure you're installing the latest packages for your image

Also, FYI in more recent versions of ubuntu - `apt-get` is deprecated in favour of `apt`, so you would just run `apt update` and `apt install -y blah` (But this may not work if the image is on an older version)